Marlin/src/core/types.h

```diff
@@ -274,6 +274,13 @@ enum AxisEnum : uint8_t {
 #define LOOP_DISTINCT_AXES(VAR) for (uint8_t VAR = 0; VAR < DISTINCT_AXES; ++VAR)
 #define LOOP_DISTINCT_E(VAR) for (uint8_t VAR = 0; VAR < DISTINCT_E; ++VAR)
 
+// Time, ticks, cycles
+typedef uint32_t cycles_t;  // Cycle counting may be larger than timer size
+typedef uint32_t ticks_t;   // Tick counting, according to each timer's rate
+typedef uint32_t micros_t;  // Microseconds
+typedef uint32_t nanos_t;   // Nanoseconds
+typedef uint32_t hertz_t;   // Hertz
```

Comment:
To expand on why I personally don't like these: visual clutter, slightly harder to process mentally since they're unfamiliar (and will be to anyone contributing), duplication of information as we already have _ns and _us endings. Particular to `ticks_t`, now there's multiple tick types counting at different rates. The compiler doesn't help us by giving warnings when we directly add/subtract differing units together. Now the code can be wrong / inconsistent in more ways (using uint32_t when should have used micros_t for example). The size and being unsigned is slightly hidden at point of use which makes it easier to overlook potentially problematic automatic int->uint conversions.